src/cpprealm/experimental/managed_dictionary.hpp

```diff
@@ -429,8 +470,22 @@ namespace realm::experimental {
         using managed<std::map<std::string, T>>::managed_base::operator=;
 
         [[nodiscard]] std::map<std::string, T> value() const {
-            // unused
-            return std::map<std::string, T>();
+            if constexpr (std::is_pointer_v<T>) {
+                throw std::runtime_error("value() is not available on collections of managed objects. Access each object via subscript instead.");
```

Comment:
Why not?

Reply:
`T` would be `MyObject*` where it needs to be `managed<MyObject*>`. Either we expose a different function for managed objects in collections or refactor value() to be able to return a different type.